Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -347,6 +348,54 @@ Value ArrayPrototype::index_of(Interpreter& interpreter)
     return Value(-1);
 }
 
+Value ArrayPrototype::reduce(Interpreter& interpreter)
+{
+    auto this_value = interpreter.this_value();
+    auto* this_object = this_value.to_object(interpreter);
+    if (!this_object)
+        return {};
+
+    auto initial_length = get_length(interpreter, *this_object);
+    if (interpreter.exception())
+        return {};
+
+    size_t start = 1;
```

Comment:
We can't just assume 1 as start if the initial value is missing, it needs to be a bit more complex than that - consider the following:

```js
> [,].reduce(() => {})
<empty>
```

i.e. an array with one element, an empty value - which we would expose.

Spec says:

```
If initialValue is present, then
    Set accumulator to initialValue.
Else,
    Let kPresent be false.
    Repeat, while kPresent is false and k < len,
        Let Pk be ! ToString(k).
        Set kPresent to ? HasProperty(O, Pk).
        If kPresent is true, then
            Set accumulator to ? Get(O, Pk).
        Set k to k + 1.
    If kPresent is false, throw a TypeError exception.
```